### PR TITLE
Swagger UI / drf spectacular added

### PR DESCRIPTION
**Pull request by to-sta**
### Contributor checklist

- [x] This pull request is on a [separate branch](https://docs.github.com/en/get-started/quickstart/github-flow) and not the main branch

---

### Description

In this PR, i have implemented the [drf-spectacular](https://drf-spectacular.readthedocs.io/en/latest/) package. 

There are two additional endpoints now:

v1/schema/ -> generates the schema.yml file (req. endpoint)
v1/schema/swagger-ui/ -> SwaggerUI for viewing and testing our API.

The schema can be generated via terminal `python manage.py spectacular --color --file schema.yml`, v1/schema/ endpoint and potentially in docker-compose. 

I have tested this on a different project and it worked like a charm. Prerequisite for testing this here are the PR's #358 #368.

### Related issue

- #294

### CODE REVIEW

File: backend/backend/settings.py [hunk 1]
```diff
@@ -140,3 +141,14 @@
 DEFAULT_AUTO_FIELD = "django.db.models.BigAutoField"
 
 SECRET_KEY = get_random_secret_key()
+
+REST_FRAMEWORK = {
+    "DEFAULT_SCHEMA_CLASS": "drf_spectacular.openapi.AutoSchema",
+}
+
+SPECTACULAR_SETTINGS = {
+    "TITLE": "Activist.org API",
```
Comment (andrewtavis) [inline backend/backend/settings.py:150]: Should be `activist.org` :)

File: backend/backend/settings.py [hunk 2]
```diff
@@ -140,3 +141,14 @@
...
+    "DESCRIPTION": "Open-source, nonprofit activism platform",
+    "VERSION": "1.0.0",
```
Comment (andrewtavis) [inline backend/backend/settings.py:152]: Maybe do `0.1.0` for now as we're working towards beta? We should have the API at the same version as everything else for now until we break it up? Or are we doing different versions for the backend and frontnend?

**Comment by andrewtavis** [comment]
I'm a bit confused by the merge conflict here, @to-sta 🤔 I'm assuming that you maintained the `requirements.txt` from prior commits, and the conflict says that there is no conflict at all, but then it looks like if I resolve it it's going to make the file blank?? Not sure. Worst comes to worst we resolve it and then revert any changes to the file.

**Comment by andrewtavis** [comment]
Here's the screenshot of the conflict manager:

Screenshot_2023-09-14 15 41 26

**Comment by andrewtavis** [comment]
Let me just resolve it and revert in case it does end up deleting all the lines in the requirements file :)

**Comment by andrewtavis** [comment]
Did delete them all 🤨 I'll send a quick commit after to fix it :)

**Comment by to-sta** [comment]
Thanks @andrewtavis. Let me know in case you need help

**Comment by andrewtavis** [comment]
Should be finished via the most recent commit, @to-sta, but let me know if something looks weird :)

**Comment by to-sta** [comment]
There are few missing packages that get installed with drf-spectacluar. 

Should look like this:

```txt
asgiref==3.7.2
astroid==2.15.6
attrs==23.1.0
black==23.3.0
certifi==2023.7.22
charset-normalizer==3.2.0
click==8.1.7
colorama==0.4.6
dill==0.3.7
Django==4.2.3
django-stubs==4.2.3
django-stubs-ext==4.2.2
djangorestframework==3.14.0
djangorestframework-stubs==3.14.2
drf-spectacular==0.26.4
gunicorn==20.1.0
idna==3.4
inflection==0.5.1
isort==5.12.0
jsonschema==4.19.0
jsonschema-specifications==2023.7.1
lazy-object-proxy==1.9.0
mccabe==0.7.0
mypy==1.5.1
mypy-extensions==1.0.0
packaging==23.1
pathspec==0.11.2
platformdirs==3.10.0
psycopg2-binary==2.9.5
pylint==2.17.5
pylint-django==2.5.3
pylint-plugin-utils==0.8.2
python-dotenv==1.0.0
pytz==2023.3
PyYAML==6.0.1
referencing==0.30.2
requests==2.31.0
rpds-py==0.10.3
ruff==0.0.287
sqlparse==0.4.4
tomli==2.0.1
tomlkit==0.12.1
types-pytz==2023.3.0.1
types-PyYAML==6.0.12.11
types-requests==2.31.0.2
types-urllib3==1.26.25.14
typing_extensions==4.7.1
tzdata==2023.3
uritemplate==4.1.1
urllib3==2.0.4
wrapt==1.15.0
```

You could either take this list or install and freeze. 

`pip install drf-spectacular`
`pip freeze > requirements.txt`

**Comment by andrewtavis** [comment]
I just did another commit that copies these in, @to-sta. My backend env isn't setup properly at the moment as I was using conda and now should switch to another provider as conda really isn't meant for development environments but more analytics (habit from my background). Will get my env setup next week before Code Night 🌙😊

**Pull request merged**